projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -141,16 +141,29 @@ export class LazyElementsLoaderService {
         script.type = 'module';
       }
       script.src = url;
-      script.onload = () => {
+      const onLoad = () => {
         if (afterLoadHook) {
           this.handleHook(afterLoadHook, tag)
             .then(notifier.resolve)
             .catch(notifier.reject);
         } else {
           notifier.resolve();
         }
+
+        cleanup();
```

Comment:
This is not a problem because at that point of time the `<script>` part is done so it can be removed even though we have some `.then` on line 147 right ?

Reply:
Yeah, sure. It'll just remove the event listener from the `script` element, but it doesn't affect the actual code execution:
```js
function delay() {
  return new Promise((resolve) => setTimeout(resolve, 1000));
}

const script = document.createElement('script');
script.src =
  'https://cdnjs.cloudflare.com/ajax/libs/zone.js/0.11.4/zone.min.js';

function onLoad() {
  delay().then(() => {
    console.log(
      'The promise has been resolved in 1 second tho the listened has been removed...'
    );
  });

  script.removeEventListener('load', onLoad);
}

script.addEventListener('load', onLoad);

document.head.appendChild(script);

```